src/suggestions.rs

```diff
@@ -0,0 +1,36 @@
+use rustyline::completion::Completer;
+use rustyline::highlight::Highlighter;
+use rustyline::hint::Hinter;
+use rustyline::validate::Validator;
+use rustyline::Context;
+use rustyline_derive::Helper;
+
+#[derive(Helper)]
+pub struct ShellHelper {
+    pub suggestion: String,
+}
+
+impl Completer for ShellHelper {
+    type Candidate = String;
+}
+
+impl Hinter for ShellHelper {
+    type Hint = String;
+
+    fn hint(&self, line: &str, pos: usize, _ctx: &Context<'_>) -> Option<String> {
+        if pos == line.len() && !self.suggestion.is_empty() && line.len() < self.suggestion.len() {
```

Comment:
What's the reasoning for `line.len() < self.suggestion.len()`?

Reply:
Removed

Reply:
Changed to `self.suggestion.starts_with(line)` so that if you start to type a different command because you don't want to use the suggested command then it removes the suggested command